foundations/foundation-config/src/main/java/org/apache/servicecomb/config/priority/impl/PropertyGetter.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.servicecomb.config.priority.impl;
+
+import com.netflix.config.DynamicStringProperty;
+import com.netflix.config.Property;
+import com.netflix.config.PropertyWrapper;
+
+public interface PropertyGetter<T> {
+  Property<T> getProperty(String propName, T defaultValue);
+
+  @SuppressWarnings("unchecked")
+  static <T> Property<T> getStringProperty(String propName, T defaultValue) {
+    return (Property<T>) new DynamicStringProperty(propName, (String) defaultValue);
+  }
+
+  class DynamicLongProperty extends PropertyWrapper<Long> {
+    public DynamicLongProperty(String propName, Long defaultValue) {
+      super(propName, defaultValue);
+    }
+
+    public Long get() {
+      return prop.getLong(defaultValue);
+    }
+
+    @Override
+    public Long getValue() {
+      return get();
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  static <T> Property<T> getLongProperty(String propName, T defaultValue) {
+    return (Property<T>) new DynamicLongProperty(propName, (Long) defaultValue);
```

Comment:
Maybe we can remove the annotation `@SuppressWarnings("unchecked")` like below?
```java
  static Property<Long> getLongProperty(String propName, Long defaultValue) {
    return new DynamicLongProperty(propName, defaultValue);
  }
```

Reply:
done